Lib/ctypes/util.py

```diff
@@ -96,8 +96,11 @@ def find_library(name):
     def _is_elf(filename):
         "Return True if the given file is an ELF file"
         elf_header = b'\x7fELF'
-        with open(filename, 'br') as thefile:
-            return thefile.read(4) == elf_header
+        try:
+            with open(filename, 'br') as thefile:
+                return thefile.read(4) == elf_header
+        except OSError:
+            return False
```

Comment:
I think this means to catch file does not exist and file is not readable.

Reply:
That's what'd expect to, but I want to make sure returning false on all possible `OSError` is actually the behavior we want, as it might introduce subtle bugs. For eg. I would expect a `BrokenPipeError` or `TimeoutError` to propagate, instead of not considering the file a valid library.

Looking at the usages of this function, I'd guess we only need `PermissionError`, and I think we only need it on the musl specific code path, so IMO it'd be better off there.

Anyway, I think I am being overly careful here, but would like to hear from the author if they have time. This is not blocker.

Reply:
The thinking here was: return True if we know it is an elf, return False if we know it is not or we don't know.

will current `find_library` implementation for GNU libc or *BSD raise an error in those cases?